File: C_plus/Count Primes.cpp
 int countPrimes(int n) {
        if(n <= 1) return 0;
        bool count[n+1];
        
        for( int i = 0; i<=n;i++){
            count[i] = true;
        }
        
        
        for( int i = 2; i<=n;i++){
            
            if(count[i]){
                
                for(int j = 2; i*j <= n;j++){
                    
                    count[i*j] = false;
                    
                }
                
            }
            
        }
        int cnt = 0;
        for( int i = 2; i<n;i++){
            if(count[i]){
                cnt++;
            }
        }
        
        return cnt;
        
    }

File: python/Count Primes.py
#sieve of Eratosthenes


import math

class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n<=2:
            return 0;
        prime=[0]*(n)
        for i in range(2,n):
            prime[i] = 1
        
        for i in range(2,int(math.sqrt(n))+1):
            if prime[i] == 1:
                j=i*i
                while j<n:
                    prime[j]=0
                    j += i
        cnt = 0
        for i in range(n):
            if prime[i]==1:
                cnt += 1
        return cnt
